dataset/Femnist_stats.md

```diff
@@ -0,0 +1,91 @@
+
+# Understand the Heterogeneous FL Data [[Jupyter Notebook](/dataset/Femnist_stats.ipynb) ]
+
+
+## Download the Femnist dataset and FedScale
+Follow the download instruction in `./dataset/download.sh` to download the FEMNIST dataset.
+
+```{code-cell}
+cd dataset
+bash download.sh -f 
+```
+[Install](./README.md) fedscale and import related libraries:
+
+```sh
+from torch.utils.data import DataLoader
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Make sure you have downloaded and installed FedScale
+from fedscale.core.utils.femnist import FEMNIST
+from fedscale.core.utils.utils_data import get_data_transform
+from fedscale.core.utils.divide_data import DataPartitioner
+from fedscale.core.argParser import args
+```
+
+
+## Data Loader
+
+```{code-cell}
+train_transform, test_transform = get_data_transform('mnist')
+train_dataset = FEMNIST('./dataset/femnist', dataset='train', transform=train_transform)
+test_dataset = FEMNIST('./dataset/femnist', dataset='test', transform=test_transform)
+```
+
+Partition the dataset by the `client_data_mapping` file, which gives the real-world client-level heterogeneity.
```

Comment:
Rewrite the ", which ..." part of the sentence to something that's less cryptic.